File: 프로그래머스/lv2/42885. 구명보트/구명보트.py
def solution(people, limit):
  people.sort()
  left_index = 0
  right_index = len(people) - 1
  count = len(people)
  while left_index < right_index:
    if people[left_index] + people[right_index] <= limit:
      left_index += 1
      right_index -= 1
      count -= 1
      continue
    right_index -= 1 
  return count

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    people.sort((a, b) => b - a);
    let i = 0;
    let j = people.length - 1;
    let answer = 0;
    while (i <= j) {
        if (people[i] + people[j] <= limit) {
            j--;
        }
        i++;
        answer += 1;
    }
    return answer;
}